src/Auth/AuthCookie.php

```diff
@@ -0,0 +1,106 @@
+
+		$auth_cookie      = wp_generate_auth_cookie( $user_id, $expiration, $scheme, $token );
+		$logged_in_cookie = wp_generate_auth_cookie( $user_id, $expiration, 'logged_in', $token );
+
+		do_action( 'set_auth_cookie', $auth_cookie, $expire, $expiration, $user_id, $scheme, $token ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+		do_action( 'set_logged_in_cookie', $logged_in_cookie, $expire, $expiration, $user_id, 'logged_in', $token ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+
+		if ( ! apply_filters( 'send_auth_cookies', true ) ) { // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+			return;
+		}
+
+		$samesite      = Utils::get_access_control_setting( 'loginCookieSameSiteOption', 'Lax' );
+		$cookie_domain = Utils::get_access_control_setting( 'loginCookieDomain', '' );
+
+		self::set_custom_cookie( $auth_cookie_name, $auth_cookie, $expire, PLUGINS_COOKIE_PATH, $cookie_domain, $secure, $samesite );
+		self::set_custom_cookie( $auth_cookie_name, $auth_cookie, $expire, ADMIN_COOKIE_PATH, $cookie_domain, $secure, $samesite );
+		self::set_custom_cookie( LOGGED_IN_COOKIE, $logged_in_cookie, $expire, COOKIEPATH, $cookie_domain, $secure_logged_in_cookie, $samesite );
+
+		if ( COOKIEPATH !== SITECOOKIEPATH ) {
+			self::set_custom_cookie( LOGGED_IN_COOKIE, $logged_in_cookie, $expire, SITECOOKIEPATH, $cookie_domain, $secure_logged_in_cookie, $samesite );
+		}
+	}
+
+	/**
+	 * Get the expiration time for the authentication cookie.
+	 *
+	 * @param int  $user_id  The ID of the user.
+	 * @param bool $remember Whether to remember the user.
+	 *
+	 * @return int The expiration time in seconds.
+	 */
+	private static function get_auth_cookie_expiration( int $user_id, bool $remember ): int {
+		$default_expiration = ( $remember ? 14 : 2 ) * DAY_IN_SECONDS;
+		return time() + apply_filters( 'auth_cookie_expiration', $default_expiration, $user_id, $remember ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+	}
+
+	/**
+	 * Wrapper for `set_custom_cookie` that includes SameSite attribute.
+	 *
+	 * @param string $name     The name of the cookie.
+	 * @param string $value    The value of the cookie.
+	 * @param int    $expires  The time the cookie expires.
+	 * @param string $path     The path on the server in which the cookie will be available on.
+	 * @param string $domain   The (sub)domain that the cookie is available to.
+	 * @param bool   $secure   Indicates that the cookie should only be transmitted over a secure HTTPS connection from the client.
+	 * @param string $samesite The SameSite mode for the cookie. Defaults to 'None'.
+	 */
+	// phpcs:ignore Generic.Metrics.Functions.TooManyArguments
+	private static function set_custom_cookie( string $name, string $value, int $expires, string $path, string $domain, bool $secure, string $samesite ): void {
```

Comment:
PHPstan is right here, we should use the same 3 `$name`, `$value` and `$options` params as the method we're wrapping.